### PR TITLE
nix-bundle.sh: Modernize CLI

### PR DESCRIPTION
**Pull request by haslersn**
Possible solution to #47.

TODOs:

* [ ] Update `README.md`
* [ ] What about `nix2appimage.sh`?

### CODE REVIEW

**Comment by haslersn** [comment]
Can somebody please explain to me what `extraTargets` are, or link me to an explaining resource? In this PR, I kept that functionality. But I can't really test it if I don't even know the desired effect.

**Comment by matthewbauer** [comment]
> Can somebody please explain to me what `extraTargets` are, or link me to an explaining resource? In this PR, I kept that functionality. But I can't really test it if I don't even know the desired effect.

The idea is you can put multiple paths into a tarball. For instance you might want to add ```glibcLocales``` to the Nix store.

File: nix-bundle.sh [hunk 1]
```diff
@@ -1,56 +1,165 @@
...
+
+target_path=
+extra_target_paths=
+
+for target in "${targets[@]}"
+do
+    nix build "${build_flags[@]}" "$target" || exit $?
+    if [ -z "$target_path" ]
+    then
+        target_path="$(readlink "$linkdir/result")"
```
Comment (matthewbauer) [inline nix-bundle.sh]: How does this handle multiple targets?

File: nix-bundle.sh [hunk 2]
```diff
@@ -1,56 +1,165 @@
...
 fi
 
-expr="with import <nixpkgs> {}; with import $nix_file {}; $bootstrap { target = $target; extraTargets = [ $extraTargets ]; run = \"$exec\"; }"
+# Run bootstrap function
+
+expr="(with import $nix_file {}; $bootstrap { target = $target_path; extraTargets = [$extra_target_paths ]; run = \"$bin\"; })"
+
+echo $expr
 
 out=$(nix-store --no-gc-warning -r $(nix-instantiate --no-gc-warning -E "$expr"))
```
Comment (matthewbauer) [inline nix-bundle.sh]: This should normally do the actual build, The hope is that it avoids creating ```/result``` dirs in the user's directory.

Comment (matthewbauer) [reply nix-bundle.sh]: Although it's not clear how to handle some of the new flags like ```-f```, etc. ```nix-instantiate``` does allow ```--arg``` and ```--argstr``` however.

Comment (haslersn) [reply nix-bundle.sh]: `nix build` has nicer output and it allows to use the modern nix cli. The `result` link is created in a temporary directory.

File: nix-bundle.sh (hunk 1 above)
Comment (haslersn) [reply nix-bundle.sh]: extra_target_paths contains the store paths (separated by whitespace) to the extra targets.

**Comment by matthewbauer** [comment]
@edolstra added some handling for "app"s which have a "program" attr. https://github.com/NixOS/nix/pull/2909. I am not sure if we can use that effectively here, since it is still unreleased.

**Comment by matthewbauer** [comment]
Had to revert this after seeing issues in https://github.com/matthewbauer/nix-bundle/issues/62 and https://github.com/matthewbauer/nix-bundle/issues/58. The issue is we don't have enough context to get the entire closure, resulting in paths getting dropped.

**Pull request merged**